src/diffusion_thomas_algorithm.cc

```diff
@@ -0,0 +1,354 @@
+/*
+ * Copyright 2025 compiler-research.org, Salvador de la Torre Gonzalez, Luciana
+ * Melina Luque
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *     SPDX-License-Identifier: Apache-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * This file contains a model developed under Google Summer of Code (GSoC)
+ * for the compiler-research.org organization.
+ */
+
+#include "diffusion_thomas_algorithm.h"
+#include "cart_cell.h"
+#include "hyperparams.h"
+#include "tumor_cell.h"
+#include "core/agent/agent.h"
+#include "core/diffusion/diffusion_grid.h"
+#include "core/real_t.h"
+#include "core/resource_manager.h"
+#include <cstddef>
+#include <string>
+#include <utility>
+#include <vector>
+
+namespace bdm {
+
+DiffusionThomasAlgorithm::DiffusionThomasAlgorithm(int substance_id,
+                                                   std::string substance_name,
+                                                   real_t dc, real_t mu,
+                                                   real_t resolution, real_t dt,
+                                                   bool dirichlet_border)
+    : DiffusionGrid(substance_id, std::move(substance_name), dc, mu,
+                    static_cast<int>(
```

Comment:
These static casts signal issue in the API... Maybe not in this code but BioDynaMo itself..